test/create_game_section/Lobby/lobby_widget_test.dart

```diff
@@ -1,48 +1,44 @@
 import 'dart:async';
 
-import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_test/flutter_test.dart';
-import 'package:mockito/mockito.dart';
 import 'package:niira/main.dart';
 import 'package:niira/models/game.dart';
 import 'package:niira/models/user_data.dart';
 import 'package:niira/screens/lobby.dart';
 import 'package:niira/services/database/database_service.dart';
-import 'package:niira/services/database/firestore_service.dart';
-import 'package:provider/provider.dart';
-
 import 'package:niira/services/navigation_service.dart';
+import 'package:provider/provider.dart';
 
 import '../../mocks/data/mock_games.dart';
 import '../../mocks/mock_user_data.dart';
 import '../../mocks/services/mock_auth_service.dart';
 import '../../mocks/services/mock_database_service.dart';
 
-class MockFirebaseInstance extends Mock implements FirebaseFirestore {}
-
 void main() {
   group('joining a game', () {
     testWidgets(
         'find a list of created games, tap to join ad navigate to inputPassword page',
         (WidgetTester tester) async {
-      // final _controller = StreamController<List<Game>>();
+      final _controller = StreamController<List<Game>>();
       final mockCreatedGames = MockGames().gamesToJoin;
-      // print('mock created games: $moc');
 
-      final mockFirebaseInstance = MockFirebaseInstance();
-      final firestoreService = FirestoreService(mockFirebaseInstance);
-      // pump mock data
-      // _controller.add(mockCreatedGames);
+      _controller.add(mockCreatedGames);
+      final mockDatabaseService = MockDatabaseService(controller: _controller);
+
+      // mockDatabaseService.streamOfCreatedGames.listen((event) {
+      //   print(event);
+      // });
 
       // init lobby page
       await tester.pumpWidget(
         MultiProvider(providers: [
-          Provider<DatabaseService>.value(value: firestoreService),
+          Provider<DatabaseService>.value(value: mockDatabaseService),
         ], child: MaterialApp(home: LobbyScreen())),
       );
-      // _controller.add(mockCreatedGames);
+
+      await tester.pumpAndSettle();
```

Comment:
this is actually all I did, the rest is pretty much noise

Reply:
Omg thank you!!